### PR TITLE
UIIN-2357 Added sort options in Actions for Instance/Holdings/Item

### PR DESCRIPTION
**Pull request by przemyslawturek**
Added sort options in Actions for Instance/Holdings/Item

## Issue
[UIIN-2357](https://issues.folio.org/browse/UIIN-2357)

## Video 
1. Start at default sort by title, changing in menu sorts with selected option

https://github.com/folio-org/ui-inventory/assets/128044179/db569cf4-b3db-4ae8-8937-f2e8f47e5d37

2. Selected sort options are remember between switch instance/browse/item tab

https://github.com/folio-org/ui-inventory/assets/128044179/2e2f0fde-1e2b-4656-b743-5abfc2d35075

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/FilterNavigation/FilterNavigation.js [hunk 1]
```diff
@@ -8,29 +8,36 @@ import {
...
-        </Button>
-      ))
+      Object.keys(segments).map((name) => {
+        const sort = segmentsSortBy.find(x => x.name === name).sort || '';
+
+        return (
+          <>
+            <Button
+              key={`${name}`}
+              to={`/inventory?segment=${name}${sort !== '' ? '&sort=' : ''}${sort}`}
```
Comment (mariia-aloshyna) [inline src/components/FilterNavigation/FilterNavigation.js:24]: Isn't the title default sort?

File: src/components/InstancesList/InstancesList.js [hunk 2]
```diff
@@ -251,10 +269,21 @@ class InstancesList extends React.Component {
...
+    return params.get('sort');
+  }
+
   getInitialToggableColumns = () => {
     return getItem(VISIBLE_COLUMNS_STORAGE_KEY) || TOGGLEABLE_COLUMNS;
   }
 
+  getInitialSegmentsSortBy = () => {
+    return Object.keys(segments).map(name => (
+      { name, sort:'title' }
```
Comment (mariia-aloshyna) [inline src/components/InstancesList/InstancesList.js:283]:
```suggestion
      { name, sort: SORTABLE_SEARCH_RESULT_LIST_COLUMNS.TITLE }
```

File: src/components/InstancesList/InstancesList.js [hunk 3]
```diff
@@ -369,7 +398,7 @@ class InstancesList extends React.Component {
       <SearchModeNavigation
         search={this.props.getLastBrowse()}
       />
-      <FilterNavigation segment={this.props.segment} onChange={this.refocusOnInputSearch} />
+      <FilterNavigation segment={this.props.segment} segmentsSortBy={this.state.segmentsSortBy} onChange={this.refocusOnInputSearch} />
```
Comment (BogdanDenis) [inline src/components/InstancesList/InstancesList.js:401]: Please format it so each prop is on new line:
```
<FilterNavigation
  segment={this.props.segment}
  segmentsSortBy={this.state.segmentsSortBy}
  onChange={this.refocusOnInputSearch}
/>
```

File: src/components/InstancesList/InstancesList.js [hunk 4]
```diff
@@ -594,6 +623,53 @@ class InstancesList extends React.Component {
...
+        goTo,
+        getParams,
+      } = this.props;
+
+      onToggle();
+
+      const params = getParams();
+      params.sort = event.target.value;
+
+      const segmentsSortBy = this.state.segmentsSortBy.map((key) => {
```
Comment (BogdanDenis) [inline src/components/InstancesList/InstancesList.js:638]: This is a duplicate of code from `didUpdate` - please create a separate method and call it in both places

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/components/InstancesList/InstancesList.js (hunk 2 above)
Comment (przemyslawturek) [reply src/components/InstancesList/InstancesList.js]: good point :)

File: src/components/FilterNavigation/FilterNavigation.js (hunk 1 above)
Comment (przemyslawturek) [reply src/components/FilterNavigation/FilterNavigation.js:24]: Yes there is set in initial component state. Upper code with empty sort option is for relevance where sort param not exists in link location

**Review by BogdanDenis — APPROVED** [review]

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**